### PR TITLE
workflows/tests: don't always run full CI on PR

### PR DESCRIPTION
**Pull request by dtrodrigues**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
This will allow a user to include the string "(syntax only)" in the PR title so that full CI doesn't run. Other options include a different string in the PR title or requiring a magic string in the PR body (`github.event.pull_request.body`). The magic string could also be part of the PR template and if a user checks/unchecks a box, then only the syntax checks would be run.

Motivated by bulk license updates or updates that otherwise won't change the formula but we still want to check the syntax and not manually have to stop CI.

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
The title makes sense to me but I think we need to make sure that we don't support this workflow for non-maintainers which makes me wonder if a label might be a better fit?

**Comment by dawidd6** [comment]
Yes, I'm in favor of a label too.

**Comment by dtrodrigues** [comment]
I think there's a _small_ use case for non-maintainers, see https://github.com/Homebrew/homebrew-core/issues/57933 from before I was a maintainer, but the label does seem better in general. I didn't think about the fact that a label could be added during PR creation.

Any suggestions for a label name/description/color?

**Comment by MikeMcQuaid** [comment]
> I think there's a _small_ use case for non-maintainers

I guess my point is: only maintainers should be able to make the decision to avoid a full CI run.



> Any suggestions for a label name/description/color?

Name: `skip-ci-tests` or something? Don't mind about others.

**Comment by dtrodrigues** [comment]
Hmm, most labels use spaces to separate words vs dashes, so I'll go ahead and create `skip ci tests` and update the logic here.

**Comment by dtrodrigues** [comment]
I applied the label to this PR and it looks like it worked: https://github.com/Homebrew/homebrew-core/pull/59208/checks?check_run_id=950793452

**Comment by dawidd6** [comment]
Could the label be: `CI-skip-style`? We already have a pair of `CI+*` labels. I think it would be more consistent.

**Comment by dtrodrigues** [comment]
That's a good point. Since we're actually still doing the style but skipping the full tests, maybe `CI-skip-tests`?

**Comment by Rylan12** [comment]
Would something like `CI tap_syntax only` be clearer? No strong preference as long as it's clear to maintainers.

If possible, I'd like to see the `tests` jobs either not show up at all or show up in a canceled or successful state. Here's what the CI window looks like right now:

Screen Shot 2020-08-05 at 6 07 07 PM

The `tests` show up in a way that makes them look like they are still queued which can be confusing as they won't actually run. It would nice if it was clear that they have completed (even if it's the grey "canceled" state).

**Comment by dtrodrigues** [comment]
The second row shows the "Skipped" from the CI. Rows 3-5 are from the [branch protection rules](https://github.com/Homebrew/homebrew-core/settings/branch_protection_rules/2104881) which I think stay yellow unless there's a success from CI vs a skip.

I don't think we want to falsely say that the tests passed and rather make sure a maintainer is careful prior to approving and merging these. We also don't want an approval to allow the test bot to automerge the PR accidentally.

**Comment by MikeMcQuaid** [comment]
> I don't think we want to falsely say that the tests passed and rather make sure a maintainer is careful prior to approving and merging these. We also don't want an approval to allow the test bot to automerge the PR accidentally.

Agreed. This should be an exceptional case.

File: .github/workflows/tests.yml
Comment (MikeMcQuaid) [inline .github/workflows/tests.yml:83]:
```suggestion
    if: github.event_name == 'pull_request' && ! contains(github.event.pull_request.labels.*.name, 'CI-syntax-only')
```

**Comment by Bo98** [comment]
LGTM. Reminds me of the [label proposal](https://github.com/Bo98/homebrew-core/pull/1) I had a few months back, but it didn't gain much support/traction over manually editing the workflow and manually merging.

**Comment by dtrodrigues** [comment]
I've updated the label name to `CI-syntax-only` to match the existing `CI-*` prefix and indicate what's it's doing (syntax check) vs not doing (skip tests).

**Review by MikeMcQuaid — APPROVED** [review]

**Review by dawidd6 — APPROVED** [review]

**Pull request merged**
